modules/polluxBidAdapter.js

```diff
@@ -1,97 +1,115 @@
+     */
+  buildRequests: function (validBidRequests) {
+    if (!Array.isArray(validBidRequests) || !validBidRequests.length) {
+      return [];
+    }
+    const payload = [];
+    var custom_url = null;
+    for (var i = 0; i < validBidRequests.length; i++) {
+      const bid = validBidRequests[i];
+      const request = {
+        bidId: bid.bidId,
+        zones: bid.params.zone,
+        sizes: bid.sizes
+      };
+      if (bid.bidderUrl && !custom_url) {
+        custom_url = bid.bidderUrl;
       }
-      var prebidUrl = bidderUrl + '?' + utils.parseQueryStringParameters(request_obj);
-      utils.logMessage('Pollux request built: ' + prebidUrl);
-      adloader.loadScript(prebidUrl, null, true);
+      payload.push(request);
     }
-  }
-
-  // expose the callback to global object
-  function _polluxHandler (response) {
-    // pollux handler
-    var bidObject = {};
-    var callback_id = response.callback_id;
-    var placementCode = '';
-    var bidObj = utils.getBidRequest(callback_id);
-    if (bidObj) {
-      placementCode = bidObj.placementCode;
+    const payloadString = JSON.stringify(payload);
+    // build url parameters
+    const domain = utils.getParameterByName('domain');
+    const tracker2 = utils.getParameterByName('tracker2');
+    const url_params = {};
+    if (domain) {
+      url_params.domain = domain;
+    } else {
+      url_params.domain = utils.getTopWindowUrl();
+    }
+    if (tracker2) {
+      url_params.tracker2 = tracker2;
+    }
+    // build url
+    var bidder_url = custom_url || PLX_ENDPOINT_URL;
+    if (url_params) {
+      bidder_url = bidder_url + '?' + utils.parseQueryStringParameters(url_params);
+    }
+    utils.logMessage('Pollux request built: ' + bidder_url);
+    return {
+      method: 'POST',
+      url: bidder_url,
+      data: payloadString
+    };
+  },
+  /**
+     * Unpack the response from the server into a list of bids.
+     *
+     * @param {*} serverResponse A successful response from the server.
+     * @return {Bid[]} An array of bids which were nested inside the server.
+     */
+  interpretResponse: function(serverResponse, bidRequest) {
+    if (!Array.isArray(serverResponse) || !serverResponse.length) {
+      utils.logMessage('No prebid response from polluxHandler for bid requests:');
+      utils.logMessage(bidRequest);
+      return [];
     }
-    if (bidObj && response.cpm > 0 && !!response.ad) {
-      bidObject = bidfactory.createBid(STATUS.GOOD, bidObj);
-      bidObject.bidderCode = bidObj.bidder;
-      bidObject.mediaType = response.mediaType;
-      bidObject.cpm = parseFloat(response.cpm);
-      if (response.ad_type === 'url') {
-        bidObject.adUrl = response.ad;
+    // loop through serverResponses
+    const bidResponses = [];
+    for (var b in serverResponse) {
+      var bid = serverResponse[b];
+      const bidResponse = {
+        requestId: bid.bidId, // not request id, it's bid's id
+        bidderCode: spec.code,
```

Comment:
Please drop `bidderCode` here. It's not required and in fact breaks adapter aliasing. Thanks sorry for the confusion.

Reply:
Will do, but just enlight me here :)
"bidderCode" was a required parameter before, or at least it was in the [docs](http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response) right?
Thanks for your help

Reply:
Right on both counts. We have since fixed the documentation to omit that param. Thanks